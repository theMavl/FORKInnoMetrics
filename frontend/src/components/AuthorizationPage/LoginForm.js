import React from 'react'
import { Link } from 'react-router-dom'
import { Field } from 'redux-form'
import Button from '../Button'
import Input from '../Input'
import { email, required } from '../../helpers/formValidators'
import styles from './style.css'
import Spinner from "../Spinner";

class LoginForm extends React.Component {
  onSubmit = () => {
    this.props.login()
  }

  render () {
    return (
      <form className={styles.form}
            onSubmit={this.props.handleSubmit(this.onSubmit)}
      >
        <h1 className={styles.msgTitle}>Login</h1>
        <Field
          name='email'
          component={Input}
          label='Email'
          placeholder='Email'
          type='email'
          validate={[required, email]}
          disabled={this.props.activeRequest}
        />
          {this.props.activeRequest && <Spinner/>}
        <Field
          name='password'
          component={Input}
          label='Password'
          placeholder='Password'
          type='password'
          validate={required}
          disabled={this.props.activeRequest}
        />
        <Button name='Login'
                type='submit'
                style={{width: '100%'}}
                disabled={this.props.submitDisabled}
        />
        <p className={styles.formError}>{this.props.formError}</p>
        <p className={styles.suggestion}>Don&#39;t have account yet?&nbsp;
          <Link to='/register'>Register</Link>
        </p>
      </form>
    )
  }
}

export default LoginForm