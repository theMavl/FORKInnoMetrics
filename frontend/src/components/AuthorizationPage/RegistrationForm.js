import React from 'react'
import { Link } from 'react-router-dom'
import { Field } from 'redux-form'
import Button from '../Button'
import Input from '../Input'
import { email, required } from '../../helpers/formValidators'
import styles from './style.css'
import Spinner from "../Spinner";

class RegistrationForm extends React.Component {
  onSubmit = () => {
    this.props.register()
  }

  render () {
    return (
      <form className={styles.form}
            onSubmit={this.props.handleSubmit(this.onSubmit)}
      >
        <h1 className={styles.msgTitle}>Register</h1>
        <Field
          name='email'
          component={Input}
          label='Email'
          placeholder='Email'
          type='email'
          validate={[required, email]}
          disabled={this.props.activeRequest}
        />
        <Field
          name='name'
          component={Input}
          label='Name'
          placeholder='Name'
          type='text'
          validate={required}
          disabled={this.props.activeRequest}
        />
        <Field
          name='surname'
          component={Input}
          label='Surname'
          placeholder='Surname'
          type='text'
          validate={required}
          disabled={this.props.activeRequest}
        />
        <Field
          name='password'
          component={Input}
          label='Password'
          placeholder='Password'
          type='password'
          validate={required}
          disabled={this.props.activeRequest}
        />
        <Button name='Register'
                type='submit'
                style={{width: '100%'}}
                disabled={this.props.submitDisabled}
        />
        {this.props.activeRequest && <Spinner/>}
        <p className={styles.formError}>{this.props.formError}</p>
        <p className={styles.suggestion}>Have account already?
          <Link to='/login'>Login</Link>
        </p>
      </form>
    )
  }
}

export default RegistrationForm