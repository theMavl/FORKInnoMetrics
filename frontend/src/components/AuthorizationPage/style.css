.content {
    padding: var(--page-padding);
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    /*justify-content: center;*/
    background-color: rgba(255, 255, 255, 1);
}

.msgTitle {
    margin: 0;
    height: calc(var(--cell-size));
    text-align: center;
}

.description {
    width: 100%;
    height: calc(var(--cell-size));
    margin: 0;
    /*text-align: center;*/
}

.buttonsRow {
    width: 100%;
    display: flex;
    /*justify-content: space-around;*/
}

.form {
    padding: 0 calc(var(--cell-size) / 4);
    width: calc(var(--cell-size) * 6);
    background-color: rgba(255, 255, 255, 1);
}

@media all and (max-width: 360px) {
    .form {
        width: 100%;
    }
}

.inputRow {
    height: calc(var(--cell-size) / 2);
}

.submitBtn {
    height: calc(var(--cell-size) / 2);
    width: var(--cell-size);
}

.errorMsg {
    color: rgba(255, 0, 0, 1)
}

.formError {
    composes: errorMsg;
    height: calc(var(--cell-size) / 4);
    width: 100%;
    text-align: center;
}

.suggestion {
    text-align: center;
}
