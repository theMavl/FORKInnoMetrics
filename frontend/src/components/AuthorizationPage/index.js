import React from 'react'
import { connect } from 'react-redux'
import { isInvalid, reduxForm } from 'redux-form'
import { loginRequest, logoutRequest, registerRequest } from '../../store/user/actionCreators'
import { userAuthorized } from '../../helpers/selectors'
import PageTemplate from '../PageTemplate'
import RegistrationForm from './RegistrationForm'
import LoginForm from './LoginForm'
import Button from '../Button'
import styles from './style.css'

class AuthPage extends React.Component {
  onLogout = () => {
    this.props.logout()
  }
  onStayTheSame = () => {
    this.props.history.push('/')
  }
  render(){
    if(this.props.authorized)
      return (
        <PageTemplate title='Logout confirmation'>
          <p className={styles.description}>
            You are logged in already.
            Would you like to keep the same session or
            login to another account?
          </p>
          <div className={styles.buttonsRow}>
            <Button name='Stay the same'
                    styleType='primary'
                    onClick={this.onStayTheSame}
                    style={{margin: '0 calc(var(--cell-size)/4) 0 0'}}
            />
            <Button name='Logout anyway'
                    styleType='secondary'
                    onClick={this.onLogout}
            />
          </div>
        </PageTemplate>
      )
    return (
        <main className={styles.content}>
          {this.props.match.path === '/login' ?
            <LoginForm {...this.props}/> : <RegistrationForm {...this.props}/>}
        </main>
    )
  }
}

const initialValues = {
  email: '',
  password: '',
  name: '',
  surname: ''
}

let AuthorizationPage = reduxForm({
  form: 'authorization',
  initialValues: initialValues
})(AuthPage)

AuthorizationPage = connect(
    (state) => ({
      authorized: userAuthorized(state),
      authFormState: state.form.authorization,
      formError: state.user.error,
      activeRequest: state.user.activeRequest,
      submitDisabled: isInvalid('authorization')(state) || state.user.activeRequest
    }),

    (dispatch) => ({
      logout: () => dispatch(logoutRequest()),
      login: () => dispatch(loginRequest()),
      register: () => dispatch(registerRequest())
    })
)(AuthorizationPage)

export default AuthorizationPage