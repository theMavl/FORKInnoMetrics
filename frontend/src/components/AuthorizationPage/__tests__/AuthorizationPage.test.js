import React from 'react'
import createStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import AuthorizationPage from '../index'
import TestRenderer from 'react-test-renderer'

const MockLoginForm = () => <div>Login</div>
const MockRegistrationForm = () => <div>Register</div>

jest.mock('../LoginForm', () => () => MockLoginForm())
jest.mock('../RegistrationForm', () => () => MockRegistrationForm())

describe('Test AuthorizationPage component', () => {

  it('Test shallow render AuthorizationPage component', () => {

    const mockStore = createStore([])
    const store = mockStore({})

    const wrapper = shallow(
      <Provider store={store}>
        <AuthorizationPage />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('Test renders \'Logout confirmation\' when user authenticated', () => {

    const mockStore = createStore([])
    const initialStore = {
      user: {
        authorized: true
      },
      form: {}
    }
    const store = mockStore(initialStore)

    const instance = TestRenderer.create(
      <Provider store={store}>
        <AuthorizationPage />
      </Provider>
    )

    expect(instance.root.find(element => element.children && element.children.includes('Logout confirmation')))
      .toBeDefined()
    expect(instance.toJSON()).toMatchSnapshot()
  })

  it('Test renders LoginForm when user is not authenticated', () => {

    const mockStore = createStore([])
    const initialStore = {
      user: {
        authorized: false
      },
      form: {}
    }
    const store = mockStore(initialStore)
    const props = {
      match: {
        path: '/login'
      }
    }

    const instance = TestRenderer.create(
      <Provider store={store}>
        <AuthorizationPage {...props} />
      </Provider>
    )

    expect(instance.root.find(element => element.children && element.children.includes('Login')))
      .toBeDefined()
    expect(instance.toJSON()).toMatchSnapshot()
  })

  it('Test renders RegistrationForm when user is not authenticated', () => {

    const mockStore = createStore([])
    const initialStore = {
      user: {
        authorized: false
      },
      form: {}
    }
    const store = mockStore(initialStore)
    const props = {
      match: {
        path: '/register'
      }
    }

    const instance = TestRenderer.create(
      <Provider store={store}>
        <AuthorizationPage {...props} />
      </Provider>
    )

    expect(instance.root.find(element => element.children && element.children.includes('Register')))
      .toBeDefined()
    expect(instance.toJSON()).toMatchSnapshot()
  })

})