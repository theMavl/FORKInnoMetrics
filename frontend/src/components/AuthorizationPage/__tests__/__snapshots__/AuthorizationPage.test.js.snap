// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AuthorizationPage component Test renders 'Logout confirmation' when user authenticated 1`] = `
<article>
  <div>
    <span>
      <h1>
        Logout confirmation
      </h1>
    </span>
  </div>
  <div>
    <p>
      You are logged in already. Would you like to keep the same session or login to another account?
    </p>
    <div>
      <button
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "margin": "0 calc(var(--cell-size)/4) 0 0",
          }
        }
      >
        <i
          className="material-icons"
        />
        <span>
          Stay the same
        </span>
      </button>
      <button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <i
          className="material-icons"
        />
        <span>
          Logout anyway
        </span>
      </button>
    </div>
  </div>
</article>
`;

exports[`Test AuthorizationPage component Test renders LoginForm when user is not authenticated 1`] = `
<main>
  <div>
    Login
  </div>
</main>
`;

exports[`Test AuthorizationPage component Test renders RegistrationForm when user is not authenticated 1`] = `
<main>
  <div>
    Register
  </div>
</main>
`;

exports[`Test AuthorizationPage component Test shallow render AuthorizationPage component 1`] = `<Connect(ReduxForm) />`;
